pkgs/applications/networking/mailreaders/neomutt/default.nix

```diff
@@ -47,9 +51,9 @@ stdenv.mkDerivation rec {
   # like pgpring, pgpewrap, ...
   NIX_LDFLAGS = "-lidn";
 
-  configureScript = "./prepare";
-
-  enableParallelBuilding = true;
+  postInstall = ''
+    ln -s $out/bin/neomutt $out/bin/mutt
```

Comment:
Was this decision intentionally? Any other linux distribution making this symlink?

Reply:
Well, it was done intentionally by me! ;-)

I have no idea if others are doing it, but if we apply the principle of least surprise we shouldn't just break calling ```mutt```.

Reply:
Renaming mutt to neomutt was done intentionally, [voted for](http://mailman.neomutt.org/pipermail/neomutt-devel-neomutt.org/2017-September/000535.html) and and [announced](http://mailman.neomutt.org/pipermail/neomutt-devel-neomutt.org/2017-September/000526.html) on ML and changelogs. It's really not mutt anymore, and wants to stand apart from Mutt.

Reply:
I agree that surprise is bad in this case. I am not yet convinced of a situation where I will be using `mutt` and `neomutt` on the same machine and thus would not want that symlink... Neomutt should be pretty much capable of doing whatever mutt does (more or less).
While I prefer having the compatibility (for now) how would you go about this in the long term?
As far as I can see ArchLinux (AUR) is still lacking any kind of action regarding the latest release. Gentoo has been moving the `neomutt` naming for a while now (https://gitweb.gentoo.org/repo/gentoo.git/tree/mail-client/neomutt). As far as I can tell they do not maintain any backwards compatibility in this case.

I think introducing `neomutt` as the name of the binary early is a good thing. (As has been done here since it is just a symlink)
Keeping `mutt` as an alias around on the long run is debatable. For the purpose of keeping legacy craft small dropping it (at some point) is probably good.

Is there any kind of deprecation warning we can produce when someone is calling `mutt` instead of `neomutt`? Making the `mutt` (binary / script) a wrapper around neomutt that spits out a warning?

Reply:
Archlinux now has a [package](https://www.archlinux.org/packages/community-testing/x86_64/neomutt/) in community-testing (before, only a PKGBUILD in AUR).
They also don't add a symlink or wrapper for `bin/mutt` there.

I'd agree a wrapper issuing a deprecation warning might be something useful for 17.09, but simply a symlink is not enough IMHO, as users don't know they need to upgrade to the new name.

Reply:
I've added a wrapper.

Reply:
Seems like the patch doesn't apply cleanly anymore :-/

Also, is there a reason you removed `enableParallelBuilding = true;`?

Additionally, neomutt recently added an [Autosetup-based build system](https://github.com/neomutt/neomutt/pull/745). Maybe we should also move to using this? (in a followup PR probably)…

Reply:
> Also, is there a reason you removed enableParallelBuilding = true;?
I didn't remove it - just moved it together with the other "build" options.

I'll rebase tonight against master.

Reply:
Ah, right - found it. Thanks!

Reply:
Is everybody OK with the wrapper?

Reply:
LGTM.